src/linker/implementation.py

```diff
@@ -37,7 +37,6 @@ def run(
         session: Optional["drmaa.Session"],
         runner: Callable,
         step_id: str,
-        input_data: List[Path],
```

Comment:
kind of important change is to take the "input_data" and replace it with InputSteps (or, derivatively, input bindings in the case of build_cluster) and passing those to the containers instead.